nodebuilder/p2p/p2p.go

```diff
@@ -0,0 +1,224 @@
+	// given peer.
+	Connect(ctx context.Context, pi peer.AddrInfo) error
+	// ClosePeer closes the connection to a given peer.
+	ClosePeer(id peer.ID) error
+	// Connectedness returns a state signaling connection capabilities.
+	Connectedness(id peer.ID) network.Connectedness
+	// NATStatus returns the current NAT status.
+	NATStatus() (network.Reachability, error)
+
+	// BlockPeer adds a peer to the set of blocked peers.
+	BlockPeer(p peer.ID) error
+	// UnblockPeer removes a peer from the set of blocked peers.
+	UnblockPeer(p peer.ID) error
+	// ListBlockedPeers returns a list of blocked peers.
+	ListBlockedPeers() []peer.ID
+	// Protect adds a peer to the list of peers who have a bidirectional
+	// peering agreement that they are protected from being trimmed, dropped
+	// or negatively scored.
+	Protect(id peer.ID, tag string)
+	// Unprotect removes a peer from the list of peers who have a bidirectional
+	// peering agreement that they are protected from being trimmed, dropped
+	// or negatively scored, returning a bool representing whether the given
+	// peer is protected or not.
+	Unprotect(id peer.ID, tag string) bool
+	// IsProtected returns whether the given peer is protected.
+	IsProtected(id peer.ID, tag string) bool
+
+	// BandwidthStats returns a Stats struct with bandwidth metrics for all
+	// data sent/received by the local peer, regardless of protocol or remote
+	// peer IDs.
+	BandwidthStats() metrics.Stats
+	// BandwidthForPeer returns a Stats struct with bandwidth metrics associated with the given peer.ID.
+	// The metrics returned include all traffic sent / received for the peer, regardless of protocol.
+	BandwidthForPeer(id peer.ID) metrics.Stats
+	// BandwidthForProtocol returns a Stats struct with bandwidth metrics associated with the given protocol.ID.
+	BandwidthForProtocol(proto protocol.ID) metrics.Stats
+
+	// ResourceState returns the state of the resource manager.
+	ResourceState() (rcmgr.ResourceManagerStat, error)
+
+	// PubSubPeers returns the peer IDs of the peers joined on
+	// the given topic.
+	PubSubPeers(topic string) []peer.ID
+}
+
+// module contains all components necessary to access information and
+// perform actions related to the node's p2p Host / operations.
+type module struct {
+	host      HostBase
+	ps        *pubsub.PubSub
+	connGater *conngater.BasicConnectionGater
+	bw        *metrics.BandwidthCounter
+	rm        network.ResourceManager
+}
+
+func newModule(
+	host HostBase,
+	ps *pubsub.PubSub,
+	cg *conngater.BasicConnectionGater,
+	bw *metrics.BandwidthCounter,
+	rm network.ResourceManager,
+) Module {
+	return &module{
+		host:      host,
+		ps:        ps,
+		connGater: cg,
+		bw:        bw,
+		rm:        rm,
+	}
+}
+
+// Info contains basic information about the node's p2p host/operations.
+type Info struct {
+	// ID is the node's peer ID
+	ID peer.ID `json:"id"`
+	// AddrInfo is the node's
+	AddrInfo peer.AddrInfo `json:"addr_info"`
+}
```

Comment:
AddrInfo already contains the peer.ID, so we simply return the peer.AddrInfo instead of our custom struct

Reply:
Or we can keep a struct, but remove the peer.ID, so if we ever want to add a new field, it won't be breaking